### PR TITLE
Add /price command

### LINKED ISSUE #140: Price information
Need to find new data sources for card prices, including our own caching if necessary to improve latency. Known options:

- https://yugiohprices.com/ (old Bastion)
- YGOPRODECK API
- [TCGPlayer](https://help.tcgplayer.com/hc/en-us/articles/201577976-How-can-I-get-access-to-your-card-pricing-data-)
- CardMarket
- CoolStuffInc
- eBay
- Amazon
- Troll and Toad
- Poke Order
- Card Central
- Ideal 808
- Alter Reality Games
- CoreTCG

use code AlphaKretin5 to ????

### PR DESCRIPTION
**Pull request by AlphaKretin**
Ported from https://github.com/AlphaKretin/ygoprodeck-tag-bot/blob/master/modules/price.ts

Closes #140.

### CODE REVIEW

File: src/commands/price.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		// Send out both requests simultaneously
+		const [, card] = await Promise.all([interaction.deferReply(), getCard(type, input, inputLanguage)]);
+		let end: number;
+		if (!card) {
+			end = Date.now();
+			useLocale(resultLanguage);
+			await interaction.editReply({ content: t`Could not find a card matching \`${input}\`!` });
+		} else {
+			const prices = await this.getPrice(card, vendor);
+			end = Date.now();
```
Comment (kevinlul) [inline src/commands/price.ts:150]: There's a lot of work being done after this, so this should be moved further down in the two branches to just before the first editReply each

File: src/commands/price.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		cardmarket: (price: number): string => `€${price.toFixed(2)}`,
+		coolstuffinc: (price: number): string => `$${price.toFixed(2)}`
+	};
+
+	async getPrice(card: Static<typeof CardSchema>, vendor: string): Promise<APIPrice | undefined> {
+		if (!card.name.en) {
+			// note: this is about our database, not the user input. Could clarify.
+			// unsure where, how or if to localise this error. Will it even come up?
+			throw new Error(t`Sorry, I can't find the price for a card with no English name!`);
+		}
```
Comment (kevinlul) [inline src/commands/price.ts:120]: We find out in a new issue :) like the CardMarket 0 that this API provides and the currency format localisation. Out of scope

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: src/commands/price.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+					4096
+				);
+				const embeds = profiles.map(p => {
+					const embed = new MessageEmbed();
+					embed.setTitle(t`Prices for ${card.name[resultLanguage]} - ${name}`);
+					embed.setDescription(p);
+					return embed;
+				});
+				await interaction.editReply({ embeds: [embeds[0]] }); // Actually returns void
+				embeds.shift();
```
[the remark this answers is not in the document]
Comment (AlphaKretin) [reply src/commands/price.ts:172]: I like this, but `Array<MessageEmbed>#shift()` returns `MessageEmbed | undefined`, and a guard on array length doesn't seem to change that.

[the remark this answers is not in the document]
Comment (kevinlul) [reply src/commands/price.ts:172]: I love TypeScript. I remember reading an issue with a similar flavour to this. Leave as-is then!

**Comment by AlphaKretin** [comment]
oh and of course as this is merged, we have to add the price api url to the environment

**Comment by kevinlul** [comment]
Now that I think of it, I actually don't think it needs to be there because this is tailored to a specific API and it was already in the open before: https://github.com/AlphaKretin/ygoprodeck-tag-bot/blob/master/config.json#L13

**Comment by AlphaKretin** [comment]
> Now that I think of it, I actually don't think it needs to be there because this is tailored to a specific API and it was already in the open before: https://github.com/AlphaKretin/ygoprodeck-tag-bot/blob/master/config.json#L13

It doesn't need to be secret, but it's consistent with how we handle our other APIs/URLs. I would also object to hardcoding it in the command file, as it may be used later to add price data to card embeds.

**Comment by kevinlul** [comment]
Wouldn't we move the entire `getPrice` function out then, which contains the URL?

**Comment by AlphaKretin** [comment]
> Wouldn't we move the entire `getPrice` function out then, which contains the URL?

...True. In that case, if that's how you want to do it I don't necessarily object, but I do lightly prefer the current state.

**Comment by kevinlul** [comment]
Let's inline it. It's consistent with [`/yugipedia`](https://github.com/DawnbrandBots/bastion-bot/blob/master/src/commands/yugipedia.ts#L40-L42) and [`/ygoprodeck`](https://github.com/DawnbrandBots/bastion-bot/blob/master/src/commands/ygoprodeck.ts#L41).

**Review by kevinlul — APPROVED** [review]
🚀 💸

**Pull request merged**